MapboxVisionTests/VisionManagerTests.swift

```diff
@@ -0,0 +1,161 @@
+@testable import MapboxVision
+import XCTest
+
+class VisionManagerTests: XCTestCase {
+    var visionManager: VisionManager?
+    var fatalErrorMessage: String?
+    var dependencies: VisionDependencies?
+
+    override func setUp() {
+        super.setUp()
+
+        let native = MockNative()
+
+        dependencies = VisionDependencies(
+            native: native,
+            synchronizer: MockSynchronizable(),
+            recorder: MockSessionRecorder(),
+            dataProvider: MockDataProvider(),
+            deviceInfo: DeviceInfoProvider()
+        )
+
+        self.visionManager = VisionManager(dependencies: dependencies!, videoSource: MockVideoSource())
+    }
+
+    func test_AssertOnVisionManagerDeallocWithoutDestroy() {
+        // Description:
+        // We called destroy in destructors of VisionManager, BaseVisionManager, VisionManagerNative and VisionManagerBaseNative,
+        // so VisionManagerBaseNative's destroy method called multiple times, so we tried to destroy already destroyed resources,
+        // which caused an EXC_BAD_ACCESS.
+        //
+        // Nevertheless the logic was sort of right, it should do assertFailure on deinit without destroy was called
+        //
+        // Given
+        // object of VisionManager
+        // When
+        // we deallocate it without destroy
+        // Then
+        // Should be assertion failure
+        expectAssertionFailure { [weak self] in
+            self?.dependencies?.native.destroy()
```

Comment:
It doesn't seem that this cleanup is needed.

Reply:
Yeah, since it's a mocked native, will remove